core/src/main/java/com/linecorp/armeria/server/http/dynamic/DynamicHttpService.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2016 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.http.dynamic;
+
+import com.linecorp.armeria.common.http.DefaultHttpResponse;
+import com.linecorp.armeria.common.http.HttpMethod;
+import com.linecorp.armeria.common.http.HttpRequest;
+import com.linecorp.armeria.common.http.HttpResponse;
+import com.linecorp.armeria.common.http.HttpStatus;
+import com.linecorp.armeria.server.ServiceRequestContext;
+import com.linecorp.armeria.server.http.AbstractHttpService;
+import com.linecorp.armeria.server.http.HttpService;
+
+/**
+ * An {@link HttpService} that serves dynamic contents.
+ */
+public final class DynamicHttpService extends AbstractHttpService {
```

Comment:
Add to this, I think approach 2 and 3 are too similar each other. How about replace approach 2 with approach 3? It would be much simpler and better. With this way, we can have 2 ways of building `DynamicHttpService`:

1. Create `DynamicHttpService` indirectly using `DynamicHttpServiceBuilder`.
2. Create `DynamicHttpService` directly by extending & annotating the class. (without `DynamicHttpServiceBuilder`)